### PR TITLE
storage for AutomationBot & merging Aggregator into AutomationBot

### PR DESCRIPTION
**Pull request by adamskrodzki**
https://app.shortcut.com/oazo-apps/story/5712/sc-create-separated-storage-contract-for-all-automationbot-data-make-automationbot-stateless

https://app.shortcut.com/oazo-apps/story/5761/sc-extend-the-automationbot-with-groups-strategies

### CODE REVIEW

File: scripts/common/types.ts [hunk 1]
```diff
@@ -28,6 +28,7 @@ export enum AutomationServiceName {
     CDP_MANAGER = 'CDP_MANAGER',
```
Comment (halaprix) [inline scripts/common/types.ts:28]: add single trigger grouptype in `TriggerGroupType`

File: test/automation-bot.ts [hunk 2]
```diff
@@ -119,35 +112,33 @@ describe('AutomationBot', async () => {
         )
 
         it('should fail if called not through delegatecall', async () => {
-            const tx = AutomationBotInstance.addTrigger(1, triggerType, false, 0, triggerData)
+            const tx = AutomationBotInstance.addTriggers(Math.pow(2, 16) - 1, [false], [0], [triggerData])
             await expect(tx).to.be.revertedWith('bot/only-delegate')
         })
```
Comment (halaprix) [inline test/automation-bot.ts:118]: add
```
        it('should fail when trying to call emitGroupDetails', async () => {
            const tx = AutomationBotInstance.emitGroupDetails(TriggerGroupType.CONSTANT_MULTIPLE, testCdpId, [])
            await expect(tx).to.be.reverted
        })
```
to explain existance of lock() :)

**Review by halaprix — DISMISSED** [review]

File: contracts/AutomationBot.sol [hunk 3]
```diff
@@ -20,9 +20,12 @@ pragma solidity ^0.8.0;
 
 import "./interfaces/ManagerLike.sol";
 import "./interfaces/ICommand.sol";
+import "./interfaces/IValidator.sol";
 import "./interfaces/BotLike.sol";
+import "./AutomationBotStorage.sol";
 import "./ServiceRegistry.sol";
 import "./McdUtils.sol";
+import "hardhat/console.sol";
```
[the remark this answers is not in the document]
Comment (adamskrodzki) [reply contracts/AutomationBot.sol:28]: fixed

File: test/automation-bot.ts (hunk 2 above)
Comment (adamskrodzki) [reply test/automation-bot.ts:118]: added

**Review by halaprix — APPROVED** [review]

**Pull request merged**